src/main/java/org/gradle/profiler/mutations/ClearProjectCacheMutator.java

```diff
@@ -0,0 +1,57 @@
+package org.gradle.profiler.mutations;
+
+import com.typesafe.config.Config;
+import org.apache.commons.io.FileUtils;
+import org.gradle.profiler.BuildMutator;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.function.Supplier;
+
+public class ClearProjectCacheMutator implements BuildMutator {
+    private final File projectDir;
+
+    public ClearProjectCacheMutator(File projectDir) {
+        this.projectDir = projectDir;
+    }
+
+    @Override
+    public void afterBuild(Throwable error) {
+        deleteGradleCache("project", projectDir);
+        File buildSrc = new File(projectDir, "buildSrc");
+        if (buildSrc.exists()) {
+            deleteGradleCache("buildSrc", buildSrc);
+        }
+    }
+
+    private void deleteGradleCache(String name, File baseDir) {
+        File gradleCache = new File(baseDir, ".gradle");
+        System.out.println(String.format("> Cleaning %s .gradle cache: %s", name, gradleCache));
+        if (!gradleCache.exists()) {
+            throw new IllegalArgumentException(String.format(
+                "Cannot delete .gradle %s directory (%s) since it does not exist",
+                name,
+                gradleCache
+            ));
+        }
+        try {
+            FileUtils.deleteDirectory(gradleCache);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName();
+    }
+
+    public static class Configurator implements BuildMutatorConfigurator {
+        @Override
+        public Supplier<BuildMutator> configure(Config scenario, String scenarioName, File projectDir, String key) {
+            boolean enabled = scenario.getBoolean(key);
+            return () -> enabled ? new ClearProjectCacheMutator(projectDir) : BuildMutator.NOOP;
+        }
```

Comment:
Can we handle the `enabled` state in a more general manner, so we don't have to duplicate this code?

Reply:
I was wondering if we need `enabled` at all. Something like this in the scenario file should be enough:
```
default {
    clear-project-caches
}
```

Same for the build cache logging mutator. WDYT?

Reply:
That could work, too. The only use of `enabled = false` would be if we had some elaborate inheritance scheme where it would make sense to enable things on top and disable them below.

Another angle to look at is if we want the configuration to be more declarative or imperative. I'd say what you are proposing takes us towards imperative.

I don't feel strongly either way.

Reply:
I don't see leaving out the boolean value of the property as moving towards less declarative. The declaration `clear-project-caches` means to add the given mutator which requires no configuration options. The other mutators don't have enable/disable options either (e.g. `apply-abi-change`).

Reply:
Looks like you can't use
```
default {
    clear-project-caches
}
```

in HOCON, so I introduced a base class for the mutators without options.